### PR TITLE
Add diagnostic transactions

### PR DESCRIPTION
**Pull request by CodaFi**

### CODE REVIEW

File: Sources/Lithosphere/DiagnosticEngine.swift [hunk 1]
```diff
@@ -18,6 +18,12 @@ public final class DiagnosticEngine {
   /// The current set of emitted diagnostics.
   private(set) public var diagnostics = [Diagnostic]()
 
+  /// The set of in-flight diagnostics for a particular transaction.
+  private var transactionDiagnostics = [UUID: [Diagnostic]]()
+
+  /// The current transaction.
+  private var activeTransaction: UUID?
```
Comment (CodaFi) [inline Sources/Lithosphere/DiagnosticEngine.swift]: Is UUID the right type to use here?  An Int would suffice but would require an extra private static member.

**Review by harlanhaskins — APPROVED** [review]
Oh this is a clever scheme.

File: Sources/Lithosphere/DiagnosticEngine.swift (hunk 1 above)
Comment (harlanhaskins) [reply Sources/Lithosphere/DiagnosticEngine.swift]: Why static? It can be an ivar of DiagnosticEngine.

Comment (CodaFi) [reply Sources/Lithosphere/DiagnosticEngine.swift]: Oh, silly me.  Still thinking in terms of global UIDs.

**Comment by CodaFi** [comment]
Passed Linux - and macOS before I started fiddling with `defer`.

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**